commands/command_push.go

```diff
@@ -19,11 +19,60 @@ var (
 	}
 	pushDryRun       = false
 	pushDeleteBranch = "(delete)"
+	pushObjectIDs    = false
 	useStdin         = false
 
 	// shares some global vars and functions with commmands_pre_push.go
 )
 
+func uploadsBetweenRefs(left string, right string) *lfs.TransferQueue {
+	// Just use scanner here
+	pointers, err := lfs.ScanRefs(left, right)
+	if err != nil {
+		Panic(err, "Error scanning for Git LFS files")
+	}
+
+	uploadQueue := lfs.NewUploadQueue(lfs.Config.ConcurrentTransfers(), len(pointers))
+
+	for i, pointer := range pointers {
+		if pushDryRun {
+			Print("push %s", pointer.Name)
+			continue
+		}
+		tracerx.Printf("checking_asset: %s %s %d/%d", pointer.Oid, pointer.Name, i+1, len(pointers))
+
+		u, wErr := lfs.NewUploadable(pointer.Oid, pointer.Name)
+		if wErr != nil {
+			if Debugging || wErr.Panic {
+				Panic(wErr.Err, wErr.Error())
+			} else {
+				Exit(wErr.Error())
+			}
+		}
+		uploadQueue.Add(u)
+	}
+
+	return uploadQueue
+}
+
+func uploadsWithObjectIDs(oids []string) *lfs.TransferQueue {
+	uploadQueue := lfs.NewUploadQueue(lfs.Config.ConcurrentTransfers(), len(oids))
+
+	for _, oid := range oids {
+		u, wErr := lfs.NewUploadable(oid, "")
```

Comment:
Agreed. However `checking_asset` doesn't seem right to me. Is is ok if I rename it to `prepare upload` ?